frontend/leads-search/lead-list/lead-search-input.component.tsx

```diff
@@ -0,0 +1,305 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import easyFetch from "../../util/easy-fetch";
+import { useCss } from "kremling";
+import { SearchParseValues } from "../../leads-search/lead-list/lead-search-dsl.helpers";
+import {
+  parseSearch,
+  SearchParse,
+  deserializeSearch,
+  allowedSearchFields,
+  serializeSearch
+} from "./lead-search-dsl.helpers";
+
+export default function LeadSearchInput(props: LeadSearchInputProps) {
+  const scope = useCss(css);
+  const [showingAdvancedSearch, setShowingAdvancedSearch] = React.useState(
+    false
+  );
+  const [search, dispatchSearch] = React.useReducer(
+    searchReducer,
+    getInitialSearch()
+  );
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const [programs, setPrograms] = React.useState([]);
+  const [events, setEvents] = React.useState([]);
+
+  React.useEffect(() => {
+    inputRef.current.setCustomValidity(search.parseResult.errors.join(". "));
+  }, [search]);
+
+  React.useEffect(() => {
```

Comment:
React effects should return a function that cancels the effect. In this case that should be `() => abortController.abort()`

Additionally, it's better practice to not have a "mount effect" that is trying to be the same thing as componentDidMount. Instead, you should treat each api request / effect as a separate effect. In other words, this effect should be split into two - one for fetching services and one for fetching events.